### PR TITLE
Improve warning when rendering objects (see #4840)

### PR DESCRIPTION
**Pull request by glenjamin**
I'm wondering whether it might be better to just use `JSON.stringify` on the object causing the warning?

### CODE REVIEW

File: src/shared/utils/__tests__/traverseAllChildren-test.js [hunk 1]
```diff
@@ -476,4 +476,45 @@ describe('traverseAllChildren', function() {
...
+      <div>
+        {new Date('2015-09-12T17:00:00Z')}
+      </div>
+    );
+
+    try {
+      traverseAllChildren(instance.props.children, function() {});
+      expect("here").toBe("not reached");
+    } catch (ex) {
+      // Can't fuzzy match on error with toThrow
```
Comment (zpao) [inline src/shared/utils/__tests__/traverseAllChildren-test.js]: You can pass a regex to `toThrow` to match.

Comment (glenjamin) [inline src/shared/utils/__tests__/traverseAllChildren-test.js]: Oh, neat! I should have just tried it instead of poking through docs :)

Comment (glenjamin) [inline src/shared/utils/__tests__/traverseAllChildren-test.js]: I tried this in https://github.com/glenjamin/react/commit/7bb820fa55011388ff1dcc8ec49c44f76f22ccde - as far as I can tell this is a Jasmine 2.0 feature that isn't available the vendored jasmine 1.3 jest contains.

**Comment by glenjamin** [comment]
Have updated the PR to pass the linter - although I'm now seeing a bunch of `no-catch-shadow` warning which weren't there before - and also seem to be false positives.

**Comment by glenjamin** [comment]
Closing since https://github.com/facebook/react/issues/4951 has now merged and does basically the same thing.

**Comment by spicyj** [comment]
Sorry, I missed this!

**Pull request closed without merging**